### PR TITLE
feat: made functions for all the api calls

### PR DESCRIPTION
**Pull request by bhtibrewal**
wrote functions for all the api calls
pages to avoid 
- [ ] videos.js
- [ ] categories.js
- [ ] index.js

### CODE REVIEW

File: src/services/data/fetchCategories.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+
+export const fetchCategories = async ({ videoStateDispatch }) => {
+    try {
+        const res = await axios.get("/api/categories");
+        videoStateDispatch({ type: "ADD_CATEGORIES", payload: res.data.categories });
```
Comment (pankaj-prog) [inline src/services/data/fetchCategories.js:6]: you should check the status first and then run the dispatch method, also follow this in the other api calls

File: src/services/like_video/likeVideo.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios"
+
+export const likeVideo = async ({ video,videoStateDispatch }) => {
+    try {
+        const res = await axios.post("/api/user/likes", { video });
+        // console.log(res.data)
```
Comment (pankaj-prog) [inline src/services/like_video/likeVideo.js:6]: remove this commented code

**Review by pankaj-prog — APPROVED** [review]

File: src/services/data/fetchCategories.js (hunk 1 above)
Comment (kgajriaAgarwal) [inline src/services/data/fetchCategories.js:6]: Should maintain the checks for various api status code

File: src/services/like_video/likeVideo.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+       videoStateDispatch({ type: "HANDLE_LIKES", payload: res.data.likes });
```
Comment (kgajriaAgarwal) [inline src/services/like_video/likeVideo.js:7]: try using optional chaining like res?.data?.likes

File: src/services/like_video/unlikeVideo.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios"
+
+export const unlikeVideo = async({_id, videoStateDispatch})=>{
+    try{
+        const res= await axios.delete(`/api/user/likes/${_id}`);
+        videoStateDispatch({ type: "HANDLE_LIKES", payload: res.data.likes });
```
Comment (kgajriaAgarwal) [inline src/services/like_video/unlikeVideo.js:6]: can use optional chaining here also

**Review by kgajriaAgarwal — APPROVED** [review]
nice modular code. Its is readable and understandable as well . good can be merged

File: src/services/data/fetchCategories.js (hunk 1 above)
Comment (bhtibrewal) [reply src/services/data/fetchCategories.js:6]: thanks for the suggestion, done

Comment (bhtibrewal) [reply src/services/data/fetchCategories.js:6]: thanks, done

File: src/services/like_video/likeVideo.js (hunk 2 above)
Comment (bhtibrewal) [reply src/services/like_video/likeVideo.js:6]: done

**Pull request merged**